CONTRIBUTING.md

```diff
@@ -325,6 +326,84 @@ if you change the history (rebase, squash, amend), and use git push --force to
 update the branch on your fork. The pull request points to that branch, not to
 specific commits.
 
+## Exercism.io Web Structure
+
+### Intro
+Exercism is built with the [Sinatra](https://github.com/sinatra/sinatra) web framework.
+If you haven't heard of it, definitely check it out. It's an awesome lightweight web frameworks for Ruby.
+
+Exercism seems to follow what some may call an MVP (Model-View-Presenter) architecture.
+Read more about that [here](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93presenter).
+
+### Base Layout (Where the Magic Happens)
+
+#### App (Routes, Presenters and Views)
+App is the user facing side of exercism.
+It handles logging in (and other functions), making calls out to the database to store or get information,
+structuring HTML (that can later be styled), etc.
```

Comment:
It technically shouldn't talk to the database much, but that's probably beside the point, since it _does_.

The routes handle requests on the exercism.io website--things like logging in, viewing exercises, profiles, and teams, and updating account settings. Most of the routes define actions that either respond with HTML directly, or redirect to an action that responds with HTML.

The HTML is defined in `erb` files in the `views`.

The Presenters do all sorts of messy things that have gotten out of hand. We're working on cleaning that up.